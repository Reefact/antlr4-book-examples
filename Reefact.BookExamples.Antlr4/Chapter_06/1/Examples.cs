#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._1 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        public void Tree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("data.csv", 6, 1);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string listStyleTree = grun.ToLispStyleTree();
            // Verify
            Approvals.Verify(listStyleTree);
        }

        [Fact]
        public void Tokens() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("data.csv", 6, 1);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

        [Fact]
        public void MermaidTree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("data.csv", 6, 1);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

    }

}