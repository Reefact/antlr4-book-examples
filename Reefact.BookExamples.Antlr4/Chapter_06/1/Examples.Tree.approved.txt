(file (hdr (row (field Details) , (field Month) , (field Amount) \r \n)) (row (field Mid Bonus) , (field June) , (field "$2,000") \r \n) (row field , (field January) , (field """zippo""") \r \n) (row (field Total Bonuses) , (field "") , (field "$5,000") \r \n))